cmd/ethermintd/ledger.go

```diff
@@ -0,0 +1,145 @@
+package main
+
+import (
+	"fmt"
+	"math/big"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/accounts/usbwallet"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/tharsis/ethermint/ethereum/rpc/backend"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/tendermint/libs/log"
+	rpctypes "github.com/tharsis/ethermint/ethereum/rpc/types"
+)
+
+const (
+	flagIndex = "index"
+)
+
+func runAddCmd(cmd *cobra.Command, _ []string) error {
+	bip44Index, _ := cmd.Flags().GetUint32(flagIndex)
+	ledgerHub, detectLedgerErr := usbwallet.NewLedgerHub()
+	if detectLedgerErr != nil {
+		return fmt.Errorf("ledger detect error %v", detectLedgerErr)
+	}
+	allLedgerWallets := ledgerHub.Wallets()
+	firstLedgerWallet := allLedgerWallets[0]
+	openWalletErr := firstLedgerWallet.Open("")
+	if openWalletErr != nil {
+		return fmt.Errorf("ledger open error %vn", openWalletErr)
+
+	}
+
+	// bip44(44), coin type(60), account, change ,index
+	hdPath := []uint32{0x80000000 + 44, 0x80000000 + 60, 0x80000000 + 0, 0, bip44Index}
+	derivedAddress, deriveErr := firstLedgerWallet.Derive(hdPath, true)
+	if deriveErr != nil {
+		return fmt.Errorf("derive address error %v", deriveErr)
+	}
+
+	bech32Address, _ := sdk.Bech32ifyAddressBytes(sdk.GetConfig().GetBech32AccountAddrPrefix(), derivedAddress.Address[:])
+	fmt.Printf("Ledger Address Index= %d\nAddress= %s  %s\n", bip44Index, derivedAddress.Address.Hex(), bech32Address)
+	closeWalletErr := firstLedgerWallet.Close()
+	if closeWalletErr != nil {
+		return fmt.Errorf("ledger close error %v", closeWalletErr)
+	}
+	return nil
+}
+
+func ledgerCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:                        "ledger",

```

Comment:
I don't understand why are we creating a new `"ledger"` command... Why can't we create a custom `keys add --ledger` instead?

Reply:
current cosmos ledger only working with cosmos ledger app , cannot use override

Reply:
so in this pr, cosmos ledger is just turned off , and added eth style ledger.
this is very simpler, no side effects for cosmos codes.